src/inmanta/export.py

```diff
@@ -505,6 +512,7 @@ def commit_resources(
         partial_compile: bool,
         resource_sets_to_remove: list[str],
         pip_config: PipConfig,
+        soft_delete: bool,
```

Comment:
*If* we decide to make this the responsibility of the exporter rather than `put_partial`, as suggested in another comment, I think we can even lift the logic to the `run` method.